avalanche/models/packnet.py

```diff
@@ -0,0 +1,527 @@
+"""
+Implements (Mallya & Lazebnik, 2018) PackNet algorithm for fixed-network
+parameter isolation. PackNet is a task-incremental learning algorithm that
+uses task identities to isolate parameters.
+
+Mallya, A., & Lazebnik, S. (2018). PackNet: Adding Multiple Tasks to a
+Single Network by Iterative Pruning. 2018 IEEE/CVF Conference on Computer
+Vision and Pattern Recognition, 7765-7773.
+https://doi.org/10.1109/CVPR.2018.00810
+"""
+
+import typing as t
+
+import torch
+import torch.nn as nn
+from torch import Tensor
+from torch.nn import functional as F
+from abc import ABC, abstractmethod
+from enum import Enum
+from avalanche.core import BaseSGDPlugin, Template
+from avalanche.models.dynamic_modules import MultiTaskModule
+from avalanche.models.simple_mlp import SimpleMLP
+from avalanche.training.templates.base_sgd import BaseSGDTemplate
+
+
+class PackNetModule(ABC):
```

Comment:
when you call it `Module` you mean `nn.Module`? if so, it should inherit from it. Otherwise I suggest to rename it for clarity.